src/components/Input/Input.tsx

```diff
@@ -219,6 +221,31 @@ class Input extends AutoControlledComponent<Extendable<IInputProps>, any> {
       </ElementType>
     )
   }
+  */
+
+  renderComponent({ ElementType, classes, styles, variables, rest }) {
+    const { input, onChange } = this.props
+    const inputClasses = classes.input
+
+    return (
+      <ElementType className={classes.root} onChange={onChange} {...rest}>
```

Comment:
note that for `onChange` event here we are not providing the callback's signature that is advertised in prop types (namely, `(e, props) => ...`): here we are not providing any component's `props` to the callback